Don't specialize min/max

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151348
* __->__ #151347

address https://github.com/pytorch/pytorch/issues/149635
Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

tugsbayasgalan committed:
Don't specialize min/max



Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't specialize min/max"


Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't specialize min/max"


Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't specialize min/max"


Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Don't specialize min/max"


Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka commented on test/dynamo/test_repros.py:
```diff
@@ -1668,7 +1668,13 @@ def fn(cfg):
         # With unspec int, maximum computation is preserved
         self.assertExpectedInline(cnt.frame_count, """1""")
-        self.assertExpectedInline(cnt.op_count, """3""")
+        if torch._dynamo.config.automatic_dynamic_shapes:
```
nit: 
```
   if torch._dynamo.config.automatic_dynamic_shapes and not torch._dynamo.config.assume_static_by_default::
                self.assertExpectedInline(cnt.op_count, """4""")
   else:
                self.assertExpectedInline(cnt.op_count, """3""")
```

tugsbayasgalan committed:
Update on "Don't specialize min/max"


address https://github.com/pytorch/pytorch/issues/149635
Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 reviewed (approved):
lgtm assuming tests pass. IIUC this was all added due to holes in our reasoning system back in 2023 that presumably have been improved since then https://github.com/pytorch/pytorch/pull/94400.

tugsbayasgalan committed:
Update on "Don't specialize min/max"


address https://github.com/pytorch/pytorch/issues/149635
Differential Revision: [D73041489](https://our.internmc.facebook.com/intern/diff/D73041489/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D73041489).
